### PR TITLE
Fix changelog check output

### PR DESCRIPTION
**Pull request by camsteffen**
Error messages are more helpful if you can see them.

changelog: none

### CODE REVIEW

**Comment by camsteffen** [comment]
@bors r+

...as a fail test...if I can.

**Comment by camsteffen** [comment]
Still don't see the error message...

**Comment by flip1995** [comment]
ISTR that if blocks in bash get executed in a sub-shell. So you can't assign to a variable. Maybe this also influences the output. I recommend putting the changelog check in a file with a `#!/bin/bash` header and run `shellcheck` over it. If you're lucky, it'll tell you why there can't be any output.

Just a wild guess though. Writing bash code is also just patchwork for me...

**Comment by camsteffen** [comment]
Got it. It was silently failing at `grep` because `pipefail` is enabled. Will add a changelog so this can be reviewed/merged.

File: .github/workflows/clippy_bors.yml [hunk 1]
```diff
@@ -34,15 +34,19 @@ jobs:
...
-          sed "s/changelog: //g")
+        body=$(curl -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" -s "https://api.github.com/repos/rust-lang/rust-clippy/pulls/$PR" | \
+          python -c "import sys, json; print(json.load(sys.stdin)['body'])")
+        output=$(grep "^changelog: " <<< "$body" | sed "s/changelog: //g") || {
+          echo 'ERROR: Could not find "changelog: ..." in the PR body' >&2
+          exit 1
+        }
         if [[ -z "$output" ]]; then
-          echo "ERROR: PR body must contain 'changelog: ...'"
+          echo "ERROR: PR body must contain 'changelog: ...'" >&2
```
Comment (flip1995) [inline .github/workflows/clippy_bors.yml:44]: This should now be unreachable.

**Review by flip1995 — COMMENTED** [review]
You should use `bors try` for PRs like this, to avoid accidentally merging stuff. It does the same as r+, but will never merge.

**Comment by camsteffen** [comment]
@bors try

Seeing what happens with a blank "changelog: ".

**Comment by flip1995** [comment]
@bors r+

Thanks!

**Pull request merged**